src/ProductDetails.js

```diff
@@ -0,0 +1,104 @@
+import styled from "styled-components";
+import { useParams } from "react-router-dom";
+import { useProductById } from "./utils/hooks/useProductById";
+import { useProductCategories } from "./utils/hooks/useProductCategories";
+import Slideshow from "./Slider/Slideshow";
+
+const ShopcartButton = styled.button`
+  display: block;
+  margin-left: auto;
+  margin-right: auto;
+  padding: 5px 20px 5px 20px;
+  background-color: rgba(0, 150, 255, 0.8);
+  font-size: 20px;
+  font-weight: bold;
+  text-decoration: none;
+  color: black;
+  border-radius: 15px;
+
+  &:hover {
+    background-color: rgba(0, 150, 255, 0.2);
+  }
+`;
+
+const ProductLabel = styled.label`
+  display: block;
+`;
+
+const ProductDetails = () => {
+  const params = useParams();
+
+  const { data: categoriesData, isLoading: isLoadingCategories } =
+    useProductCategories();
+
+  const {
+    data: { results },
+    isLoading: isLoadingProduct,
+  } = useProductById(params.productId);
+
+  let productSlides = [];
+  let product = null;
+
+  let category = "";
+
+  if (isLoadingProduct === false) {
+    let [{ data: productData }] = results;
+
+    product = productData;
+
+    productSlides = productData.images.map(({ image }) => ({
+      text: "",
+      url: image.url,
+      alt: image.alt,
+    }));
+
+    if (isLoadingCategories === false) {
+      let productCategory = categoriesData.results.find(
+        (category) => category.id === productData.category.id
+      );
+      category = productCategory.data.name;
+    }
+  }
+
+  return (
+    <>
+      {!isLoadingProduct && (
+        <>
+          <Slideshow slides={productSlides}></Slideshow>
+          <ProductLabel>{product.name}</ProductLabel>
+          <ProductLabel>{product.price}</ProductLabel>
+          <ProductLabel>SKU: {product.sku}</ProductLabel>
+          <ProductLabel>Category: {category}</ProductLabel>
+          <ProductLabel>
+            Labels:
+            {results[0].tags.map((tag, idx) => (
+              <label key={idx}>{tag},</label>
+            ))}
+          </ProductLabel>
+          <p>{product.description[0].text}</p>
```

Comment:
Todo esto podría ser su propio componente y como creo que solo va a ser utilizado aquí en esta página de `ProductDetaiils` podrías dejarlo como hijo de este componente, es decir, una carpeta andidada dentro de la carpeta de este mismo.

Tienes una coma que siempre sale en la categoría final podrías cambiar esto si haces lo siguiente.
```suggestion
          <ProductLabel>{product.name}</ProductLabel>
          <ProductLabel>{product.price}</ProductLabel>
          <ProductLabel>SKU: {product.sku}</ProductLabel>
          <ProductLabel>Category: {category}</ProductLabel>
          <ProductLabel>
            Labels: { results[0].tags.join(', ') }
          </ProductLabel>
          <p>{product.description[0].text}</p>
```

Así solo las uniones del arreglo tendrán la coma, claro que con esto pierdes la posibilidad de redirigir a categorías en el futuro como si fueran links porque ya no es cada una su propio componente. Cuando esto sea necesario sugiero que entonces crees un componente que contenga varias etiquetas, allí mismo podrías implementar una lógica más robusta y le quitamos responsabilidad a este componente padre.